lib/features/pokemon_overview/pokemon_overview_page.dart

```diff
@@ -20,15 +21,29 @@ class PokemonOverviewPage extends StatelessWidget {
         centerTitle: true,
         backgroundColor: primaryColor,
       ),
-      body: GridView.builder(
-        padding: const EdgeInsets.all(10.0),
-        itemCount: pokemons.length,
-        gridDelegate: const SliverGridDelegateWithFixedCrossAxisCount(crossAxisCount: pokemonColumnCount),
-        itemBuilder: (_, index) {
-          final pokemon = pokemons[index];
-          return PokemonCard(pokemon: pokemon);
+      body: pokemons.when(
+        loading: () => const Center(child: CircularProgressIndicator()),
+        error: (errorMessage) {
+          WidgetsBinding.instance.addPostFrameCallback((_) {
```

Comment:
u can use fat arrow notation if this won't overlap the vertical suggested line 🙂